### DISCUSSION TITLE
Textbox not entering "true" editing mode

### DISCUSSION THREAD

**Discussion by SupremeDeity** [fabricjs/fabric.js#10118] [Q&A] [answered]
Not sure if this is a possible bug or its supposed to happen. 
I am using Nuxt(vue) + fabricjs 6.4. When i do something like this:

```js
 const pointer = fabricCanvas.getScenePoint(o.e);
    const text = new fabric.Textbox("Edit text", {
      left: pointer.x,
      top: pointer.y,
      fontFamily: "Kalam",
      fontSize: 20,
      fill: "white",
      editable: true,
      selectable: true,
      width: 100,
    });
    text.selectAll();
    text.enterEditing();
    fabricCanvas.add(text);
    fabricCanvas.requestRenderAll();
```

**Problem is**: The hidden textbox appears from what i can see in the devtools, The text gets selected and the editing cursor appears but i cant really edit the text at all. The only solution i found to this is to use `activeOn: "up"` which gives the correct behavior. Is this a possible bug? There's a bit of code in my application but i doubt anything is affecting this. I will try create a reproducible minimum example of this when i get the time.

**Comment by asturur** [comment]
i think you should do:
```ts
    const text = new fabric.Textbox("Edit text", {
      left: pointer.x,
      top: pointer.y,
      fontFamily: "Kalam",
      fontSize: 20,
      fill: "white",
      editable: true,
      selectable: true,
      width: 100,
    });
    fabricCanvas.add(text);
    fabricCanvas.setActiveObject(text);
    text.enterEditing();
    text.selectAll();
    fabricCanvas.requestRenderAll();
 ```

this would be an order of operation that mimic user actions and should work out of the box. If this does not work then there is a bug

**Reply by SupremeDeity** [reply]
Ok i will take a look whether this order of operation works or not whenever i get the time again.

**Comment by SupremeDeity** [comment]
@asturur just tested this out. It doesn't work. Unless i remove `fabricCanvas.setActiveObject(text);` and have `activeOn: "up"`

**Reply by asturur** [reply]
well it should. i ll try myself and add to this discussion.

**Reply by asturur** [reply]
oh wait are you doing this on some kind of click action?

**Reply by SupremeDeity** [reply]
Yes. I click to add the textbox to my mouse position.

**Reply by asturur** [reply, marked as answer]
The issue with that is that you are probably stealing focus to the textbox.
Try to move your code to mouseUP event and if you can't do that or if that doesn't work do that part of the textbox code ( creation, position, and all the rest ) in a setTimeout of few ms ( 2-4 are enough ) to be sure your code execute after the events.

**Reply by SupremeDeity** [reply]
Alright. I guess its not a bug. I get what you mean now. activeOn: "up" works fine for now so i guess I'll keep it as it is, but if i face issues I'll do that setTimeout.

**Reply by asturur** [reply]
activeOn: up  and using mouseUp are 2 different things, is that clear to you?
I don't want to confuse you

**Reply by SupremeDeity** [reply]
> activeOn: up  and using mouseUp are 2 different things, is that clear to you?
> I don't want to confuse you

Ah yes ik that. I am just doing this for now. If it causes issues later I'll do one of your suggestions. I just didnt realize that i was stealing focus from the text somehow

**Comment by timoostrich** [comment]
For people coming from google: I had a similar problem (textarea focus was stealed) in my fabric.js web app (Fontilo) when integrating Vue.

It was caused by VueFinalModal which keeps the focus. Settings focus-trap to false `:focus-trap="false"` fixed that.
I used VueFinalModal for some dynamic sidebars and finding that took me hours.

**Reply by asturur** [reply]
This should be added in JSDOCS somewhere. or maybe a dedicated page 'working with text in apps' on the website
